internal/kafka/internal/services/kafka.go

```diff
@@ -931,7 +943,7 @@ type ClusterSelection struct {
 	SupportedInstanceType string
 }
 
-func (k *kafkaService) CountStreamingUnitByRegionAndInstanceType() ([]KafkaStreamingUnitCountPerRegion, error) {
+func (k *kafkaService) CountStreamingUnitByRegionAndInstanceType() (KafkaStreamingUnitCountPerRegionList, error) {
```

Comment:
This returns all Kafkas, no matter the status, as long instance type, size id and cluster id is set. Since I'm using this in the FirstReadyCluster strategy, this means Kafkas that have already been deleted from the Data Plane is still counted as taking up capacity in that cluster. Only until this Kafka is deleted from the Fleet Manager will that capacity be freed.

For context, once the Kafka has been deleted in the Data Plane, Fleetshard will update its status from `deprovision` to `deleting`. In the `deleting` phase, only quota from AMS needs to be removed then it's deleted in the Fleet Manager.

Would these Kafkas in a *deleting* state be okay to include in the capacity used? This function is also used for calculating the capacity_used metrics. What should be considered as a *used* cluster capacity?

Reply:
Very good point. 

What is done in this PR (including the `deleting` and `deprovision`) is no different from what was done for the manual cluster placement strategy which calls https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/7c9fb1601a64322f38936aaadb70d1c9876b3b7f/internal/kafka/internal/services/clusters.go#L392 at least we do not have an inconsistency ;-) 

But you are making a good case here and we probably should stop counting the `deleting` kafka in capacity used in a cluster since they've been removed from the data plane already.

What do others think? @miguelsorianod @VanillaSpoon @JameelB 

Also cc @pb82 and @davidffrench

Reply:
+1 to not count clusters in `deleting` state as occupying resources

Reply:
+1 in that case

Reply:
@JameelB can we create a follow up for this discussion so that we do not forget about it? Thanks

Reply:
jira created: https://issues.redhat.com/browse/MGDSTRM-9252